### DISCUSSION TITLE
Discuss: troubleshooting / common problems page for dev env

### DISCUSSION THREAD

**Discussion by starfetch** [freesewing/freesewing#1208] [Forum]
The idea was raised on discord that collecting possible problems that occur during installation / usage of the dev environment would be a good idea, so that future users that run into the same problem have an idea of what to do. 

We could add this as a section on freesewing.dev, maybe separated into the three operating systems.

This discussion is for collecting ideas on how to implement it, and to collect tips in the meantime.

**Comment by bobgeorgethe3rd** [comment]
**Solution to Missing Temp Folder**
OS: Windows
Error: ENOENT: no such file or directory, lsat 'C:\Users\%YOUR USERNAME%\Appdata\Local\Temp;c:'
If an error like or similar to this appears it means the script is unable to create temp files.
This is generally because the temp path directory is incorrect in your Enviorment Variables.
But don't worry the fix is quite simple
1. Open File Explorer
2. Right Click "This PC"
3. Left Click "Properties", This will open settings.
4. Left Click "Advanced system settings" This will open the Advanced Tab of System Properties
5. Left Click "Enviorment Variables..."
6. In "User variables for %YOUR USERNAME%" Left Click "TEMP", This will hightlight it
7. Left Click the "Edit" underneath "User variables for %YOUR USERNAME%"
8. Change "Variable Value" to "C:\Users\%YOUR USERNAME%\AppData\Local\Temp" replacing %YOUR USERNAME% with your PC user name.
9. Left Click "OK"
10. Repeat steps 6-9 for TMP
11. Left Click "OK" on "Enviorment Variables"
12. Left Click "OK" on "System Properties"

This should of reset you Temp Folder Directory. You will need to close and re-open Command Prompt or equivalent for this to take affect on Command Prompt. Remember to start nvm back up before navigating to the relevant folders and trying npm start again.

**Comment by starfetch** [comment]
If you get the error ``./src/App.js` Module not found: Can't resolve './pattern/src/index.js' in '/pathTo/freesewing/packages/YourNewPattern/example/src'``, you have to create a symlink that points to `../..` in `example/src`.
On a linux machine, you could do that with `ln -s ../.. pattern` to create a `pattern` symlink.

**Reply by anna-puk** [reply]
On Windows, here's what I used:

`mklink /D "D:\repos\GitHub\freesewing\packages\ursula\example\src\pattern" "D:\repos\GitHub\freesewing\packages\ursula\"`

(I don't know whether I could have used `..` instead.)

**Comment by starfetch** [comment]
If after updating your `node.js` version you find yourself unable to run any `yarn` command (i.e. errors like `bash: yarn: command not found`), try `corepack enable`, since `yarn` might not be enabled anymore. (Looking at the [documentation of `yarn`](https://yarnpkg.com/getting-started/install/) can be helpful.)

**Reply by joostdecock** [reply]
This is documentation for yarn v2. you should use yarn v1 instead. If it's not (or no longer) available, just run:

```sh
npm install --global yarn
```

**Comment by bobgeorgethe3rd** [comment]
cannot resolve @freesewing/[insert package]

When this occurs it means you have not installed a package the pattern needs to run.
This most commonly occurs when trying to extend or install and existing pattern with the most common culprits being either needing another pattern/block for the pattern like Brian, Titan etc. or a specific plugin that is not part of the default bundle such as snapseries or gore etc. But do not fret this is a simple fix!

1. Take note of the module it can't reslove e.g @freesewing/brian
2. Close the terminal you are running the pattern in.
3. Open a new terminal and navigate to your pattern folder. DO NOT GO TO EXAMPLE!
4. Run either npm i @freesewing/[module you are installing] or yarn i @freesewing/<module you are installing> (This depends on which package you selected to create your pattern in the first place)
5. Wait for install to finish.
6. Navigate to YOUR-PATTERN/example and spin up the pattern.
7. Repeat if there are any other missing modules.
8. You're all set now have fun coding :)


This made not be the best written but it is a common one so thought good to add.

**Comment by bobgeorgethe3rd** [comment]
This is not so much for dev but it is more for when someone is using the github text editor to make changes.
Problem:
- Commit Changes button is greyed out
- Preview not accesible
- (From what I have heard) Naming suscessor not working

The issue:
The use of an adblocker tyoe extension with the most common one I have heard causing problems being Ghostery.

Temp Fix:
Pausing the exstension on github or temporarily disabling Ghostery.

Permanent Fix:
Adding "github.com" to the trusted sites of the extension.

Issue found thanks to @Duck https://discord.com/channels/698854858052075530/757631165531291771/1002735983558660238

**Comment by SirCharlotte** [comment]
Hi, I hope this is the right place to ask for help. I've tried to set up the environment to create a new pattern but I'm running into weird errors. After successfully cloning, installing `node` et al., and then running `yarn kickstart` I used the command `yarn new design` and got this error:  
```
$ yarn new design
yarn run v1.22.10
$ node --experimental-json-modules --no-warnings scripts/add-software.mjs design
node:internal/errors:477
    ErrorCaptureStackTrace(err);
    ^

TypeError [ERR_IMPORT_ASSERTION_TYPE_MISSING]: Module "file:///home/charlotte/freesewing/config/software/designs.json" needs an import assertion of type "json"
    at new NodeError (node:internal/errors:387:5)
    at validateAssertions (node:internal/modules/esm/assert:82:15)
    at defaultLoad (node:internal/modules/esm/load:84:3)
    at nextLoad (node:internal/modules/esm/loader:165:28)
    at ESMLoader.load (node:internal/modules/esm/loader:608:26)
    at ESMLoader.moduleProvider (node:internal/modules/esm/loader:464:22)
    at new ModuleJob (node:internal/modules/esm/module_job:63:26)
    at ESMLoader.#createModuleJob (node:internal/modules/esm/loader:483:17)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:441:34)
    at async ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:78:21) {
  code: 'ERR_IMPORT_ASSERTION_TYPE_MISSING'
```
Maybe someone has an idea how to solve this?

**Reply by joostdecock** [reply]
Hi there @SirCharlotte thanks for reaching out.

We've started working on the upcoming version 3 in our `develop` branch, so things are a bit volatile at the moment.

You can use the `v2` or `main` branches for more stable code.

To change branches (to `v2` for example), run:

```
git checkout v2
```

Then make sure to run `yarn kickstart` again.

FYI: there's more people active on Discord than here. It's often the best place to get support.
 👉 https://discord.freesewing.org

**Reply by SirCharlotte** [reply]
Thanks! Using `v2`, I got
```
error - ./shared/components/wrappers/workbench.js:14:0
Module not found: Can't resolve 'shared/components/workbench/export.js'
  12 | import LabDraft from 'shared/components/workbench/draft/index.js'
  13 | import LabSample from 'shared/components/workbench/sample.js'
> 14 | import ExportDraft from 'shared/components/workbench/export.js'
  15 | import GistAsJson from 'shared/components/workbench/json.js'
  16 | import GistAsYaml from 'shared/components/workbench/yaml.js'
  17 | import DraftEvents from 'shared/components/workbench/events.js'
```
so I did `cp bib/shared/components/icons/export.js bib/shared/components/workbench/`. Now it works fine.

**Reply by joostdecock** [reply]
That's not the same component though, but as long as you don't export it'll be fine 😂

I'm AFK now, but I'll look into it. Thanks for reaching out

**Reply by BenJamesBen** [reply]
I don't fully understand the issue, but for the v3 version, I note that the following change fixes it:
```diff
--- a/scripts/add-software.mjs
+++ b/scripts/add-software.mjs
@@ -6,7 +6,7 @@ import { banner } from './banner.mjs'
 import mustache from 'mustache'
 import { execSync } from 'child_process'
 // Software
-import designs from '../config/software/designs.json'
+import designs from '../config/software/designs.json' assert {type: "json"}
 
 const type = process.argv[2]
```

(I was not able to reproduce the v2 issue. I was able to switch to the v2 branch, `nvm use lts/fermium` to switch back to Node 14, `yarn kickstart`, and `yarn new design` just fine without any errors.)

**Comment by joostdecock** [comment]
> *The idea was raised on discord that...*

If I had a euro for every idea raised on Discord :joy:

Stale. Closing.
